File: rplugin/python3/iron/repls/haskell.py
# encoding:utf-8
"""Haskell repl definitions for iron.nvim. """
from iron.repls.utils.cmd import detect_fn

ghci = {
    'command': 'ghci',
    'language': 'haskell',
}

stackghci = {
    'command': 'stack ghci',
    'language': 'haskell',
}

intero = {
    'command': 'stack ghci --with-ghc intero',
    'language': 'haskell',
    'detect': detect_fn('intero'),
}

File: lua/iron/fts/haskell.lua
local haskell = {}

haskell.stack_intero = {
  command = {"stack", "ghci", "--with-ghc", "intero"},
}

haskell.stack = {
  command = {"stack", "ghci"},
}

haskell.ghci = {
  command = {"ghci"},
}

return haskell
